### DISCUSSION TITLE
How to Implement a Function to Calculate Points Based on Specific Number Conditions in JavaScript?

### DISCUSSION THREAD

**Discussion by selviast** [community/community#140064] [Programming Help]
### Body

/** 1. Calculate Points

Create a function to calculate points.

The input is an array of numbers represented as a variable arr and consists of N (which is the length of the array) followed by its elements. findTotal is a function that will calculate the total points based on:

Add 2 points for each even number.
Add 3 points for each odd number, except the number "7".
Add 6 points for each occurrence of the number "7" in the array. Return a boolean false if the array is empty.
Example 1: Input: findTotal([1,2,3]) Output: 8

Example 2: Input: findTotal([4,7]) Output: 8

NOTE:

Use return to display the output.
Provide comments explaining the purpose of the code at important lines. */

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ihwan4rfa** [comment]
You can solve the problem by iterating through the array and applying different conditions for the elements. Here's a JavaScript function that calculates the total based on the given rules: adding 6 if the number is 7, 2 if it's even, and 3 if it's odd.

function findTotal(arr) {
    let count = 0;

    for (let i = 0; i < arr.length; i++){
        count += arr[i] === 7 ? 6 : (arr[i] % 2 === 0 ? 2 : 3);
    }

    return count;
}

const input1 = [1,2,3,4,5,6,7];
const input2 = [1,2,3];
const input3 = [4,7];
console.log(findTotal(input1));
console.log(findTotal(input2));
console.log(findTotal(input3));

This function handles three test cases and applies the rules efficiently

**Comment by ebndev** [comment]
Please see [this comment](https://github.com/orgs/community/discussions/140065#discussioncomment-10798767) for reference to why this discussion was closed and locked. Thanks!
